geode-for-redis/src/main/java/org/apache/geode/redis/internal/commands/executor/set/SetOpExecutor.java

```diff
@@ -41,12 +43,36 @@ public RedisResponse executeCommand(Command command, ExecutionHandlerContext con
 
     List<RedisKey> commandElements = command.getProcessedCommandKeys();
     List<RedisKey> setKeys = commandElements.subList(setsStartIndex, commandElements.size());
-    if (isStorage()) {
-      RedisKey destination = command.getKey();
-      int storeCount = doStoreSetOp(command.getCommandType(), context, destination, setKeys);
-      return RedisResponse.integer(storeCount);
+    RegionProvider regionProvider = context.getRegionProvider();
+    try {
+      for (RedisKey k : setKeys) {
+        regionProvider.ensureKeyIsLocal(k);
+      }
+    } catch (RedisDataMovedException ex) {
+      return RedisResponse.error(ex.getMessage());
+    }
+
+    /*
+     * SDIFFSTORE, SINTER, SINTERSTORE, SUNION, SUNIONSTORE currently use the else part of the code
+     * for their
+     * implementation.
+     * TODO: Once the above commands have been implemented remove the if else and
+     * refactor so it implements doSetOp
+     */
+
+    if (command.isOfType(RedisCommandType.SDIFF)) {
+      Set<byte[]> resultSet = context.lockedExecute(setKeys.get(0), new ArrayList<>(setKeys),
```

Comment:
Could you just pass in "setKeys" instead of "new ArrayList<>(setKeys)"? setKeys is a sublist but I think it allows changes to the sublist which seems okay in this context.

Reply:
If I passed in setKeys directly in then it modifies the list and removes all the keys. @DonalEvans and I were using ZUnionStore as a guide to do this, and it was also passing a new list for the keys.

Reply:
The `lockedExecute()` method sorts the list passed to it, which results in the order of the keys in `setKeys` changing if it's passed in. Since we also pass `setKeys` into the `sdiff()` method, this results in an incorrect result for the diff, hence the need to pass a copy to one of the methods.

Reply:
Thanks for the explanation. I agree